### ISSUE TITLE
Federation issues with Pleroma

### ISSUE THREAD

**Issue by kousu** [lowercasename/gathio#81]
As a follow up to https://github.com/lowercasename/gathio/issues/35, while #79 seems to have fixed federation with Mastodon, I can't seem to get federation with Pleroma. I'm opening this to track my debugging efforts.

**Comment by kousu** [comment]
I created https://gath.io/xJU4Zqvb33EKvATHChAbV

![Screenshot 2022-11-13 at 18-19-13 sdfsdfsdfsdf · gathio](https://user-images.githubusercontent.com/987487/201549887-adc02a39-5666-4c65-8d38-0c2a1a12d26f.png)

In pleroma, I found my [`/opt/pleroma/config/prod.exs`](https://docs-develop.pleroma.social/backend/configuration/cheatsheet/) and set

```
config :logger, :console, level: :debug
```

then did

```
systemctl restart pleroma
```

and watched the logs:

```
journalctl -f -u pleroma
```

While logged in to my instance I searched for @xJU4Zqvb33EKvATHChAbV@gath.io in https://public.garden/search. It wasn't found:

![Screenshot 2022-11-13 at 18-25-10 Search Public Garden](https://user-images.githubusercontent.com/987487/201550102-b619a031-d3c1-4dbc-89b3-3edb1d08f1ae.png)


 I found this in the logs:

<details>

```
Nov 13 23:17:54 public.garden mix[24871]: 23:17:54.230 [info] GET /api/v2/search
Nov 13 23:17:54 public.garden mix[24871]: 23:17:54.231 request_id=FydIHIFp0KApXzkAAjnR [debug] GET /api/v2/search
Nov 13 23:17:54 public.garden mix[24871]: 23:17:54.231 request_id=FydIHIFp0KApXzkAAjnR [debug] Processing with Pleroma.Web.MastodonAPI.SearchController.search2/2
Nov 13 23:17:54 public.garden mix[24871]:   Parameters: %{"limit" => "20", "q" => "@xJU4Zqvb33EKvATHChAbV@gath.io", "resolve" => "true", "type" => "accounts"}
Nov 13 23:17:54 public.garden mix[24871]:   Pipelines: [:api]
Nov 13 23:17:54 public.garden mix[24871]: 23:17:54.233 request_id=FydIHIFp0KApXzkAAjnR [debug] QUERY OK source="oauth_tokens" db=0.7ms queue=0.2ms idle=1312.0ms
Nov 13 23:17:54 public.garden mix[24871]: SELECT o0."id", o0."token", o0."refresh_token", o0."scopes", o0."valid_until", o0."user_id", o0."app_id", o0."inserted_at", o0."updated_at" FROM "oauth_tokens" AS o0 WHERE (o0."token" = $1) ["kNN909hiTUU5YjRP8kBWmHTrFYv0XNOGCYWlBPbUZ8M"]
Nov 13 23:17:54 public.garden mix[24871]: 23:17:54.234 [debug] QUERY OK source="users" db=0.6ms idle=1313.8ms
Nov 13 23:17:54 public.garden mix[24871]: SELECT u0."id", u0."bio", u0."raw_bio", u0."email", u0."name", u0."nickname", u0."password_hash", u0."keys", u0."public_key", u0."ap_id", u0."avatar", u0."local", u0."follower_address", u0."following_address", u0."featured_address", u0."tags", u0."last_refreshed_at", u0."last_digest_emailed_at", u0."banner", u0."background", u0."note_count", u0."follower_count", u0."following_count", u0."is_locked", u0."is_confirmed", u0."password_reset_pending", u0."is_approved", u0."registration_reason", u0."confirmation_token", u0."default_scope", u0."domain_blocks", u0."is_active", u0."no_rich_text", u0."ap_enabled", u0."is_moderator", u0."is_admin", u0."show_role", u0."uri", u0."hide_followers_count", u0."hide_follows_count", u0."hide_followers", u0."hide_follows", u0."hide_favorites", u0."email_notifications", u0."mascot", u0."emoji", u0."pleroma_settings_store", u0."fields", u0."raw_fields", u0."is_discoverable", u0."invisible", u0."allow_following_move", u0."skip_thread_containment", u0."actor_type", u0."also_known_as", u0."inbox", u0."shared_inbox", u0."accepts_chat_messages", u0."last_active_at", u0."disclose_client", u0."accepts_email_list", u0."pinned_objects", u0."is_suggested", u0."last_status_at", u0."birthday", u0."show_birthday", u0."location", u0."language", u0."last_move_at", u0."notification_settings", u0."blocks", u0."mutes", u0."muted_reblogs", u0."muted_notifications", u0."subscribers", u0."multi_factor_authentication_settings", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."nickname" = $1) ["xJU4Zqvb33EKvATHChAbV@gath.io"]
Nov 13 23:17:54 public.garden mix[24871]: 23:17:54.316 [warning] Can't find LRDD template in "https://gath.io/.well-known/host-meta": {:ok, %Tesla.Env{__client__: %Tesla.Client{adapter: {Tesla.Adapter.Hackney, :call, [[]]}, fun: nil, post: [], pre: []}, __module__: Tesla, body: "<!DOCTYPE html>\n<html>\n\n  <head>\n    <meta charset=\"utf-8\">\n    <meta name=\"viewport\" content=\"width=device-width, initial-scale=1, shrink-to-fit=no\">\n\n    <link rel=\"apple-touch-icon\" sizes=\"180x180\" href=\"/apple-touch-icon.png\">\n    <link rel=\"icon\" type=\"image/png\" sizes=\"32x32\" href=\"/favicon-32x32.png\">\n    <link rel=\"icon\" type=\"image/png\" sizes=\"16x16\" href=\"/favicon-16x16.png\">\n    <link rel=\"manifest\" href=\"/site.webmanifest\">\n    <link rel=\"mask-icon\" href=\"/safari-pinned-tab.svg\" color=\"#5bbad5\">\n    <meta name=\"msapplication-TileColor\" content=\"#da532c\">\n    <meta name=\"theme-color\" content=\"#ffffff\">\n\n    <meta property=\"og:type\" content=\"website\">\n    <meta property=\"og:image:width\" content=\"260\">\n    <meta property=\"og:image:height\" content=\"260\">\n    <meta property=\"og:description\" content=\"An easier, quicker, and much less privacy-invading way to make and share events\">\n    <meta property=\"og:title\" content=\"\">\n    <meta property=\"og:url\" content=\"https:///\">\n    <meta property=\"og:image\" content=\"https:///og-image.jpg\">\n\n    <meta name=\"twitter:card\" content=\"summary\">\n    <meta name=\"twitter:title\" content=\"\">\n    <meta name=\"twitter:description\" content=\"An easier, quicker, and much less privacy-invading way to make and share events\">\n    <meta name=\"twitter:image\" content=\"https:///og-image.jpg\">\n\n\n    <title></title>\n\n    <!-- FONTS -->\n    <link rel=\"stylesheet\" href=\"/css/fontawesome.css\">\n\n    <!-- CSS -->\n    <link rel=\"stylesheet\" href=\"/css/bootstrap.min.css\">\n    <link href=\"/css/datepicker.min.css\" rel=\"stylesheet\" type=\"text/css\">\n    <link rel=\"stylesheet\" href=\"/css/select2.min.css\">\n    <link rel=\"stylesheet\" href=\"/css/style.css\">\n\n    <!-- JS -->\n    <script src=\"/js/jquery-3.4.1.min.js\"></script>\n    <script src=\"/js/popper.min.js\"></script>\n    <script src=\"/js/jquery.form-validator.js\"></script>\n    <script src=\"/js/bootstrap.min.js\"></script>\n    <script src=\"/js/datepicker.min.js\"></script>\n    <script src=\"/js/moment.js\"></script>\n    <script src=\"/js/jquery.uploadPreview.min.js\"></script>\n    <script src=\"/js/clipboard.min.js\"></script>\n    <script src=\"/js/autosize.min.js\"></script>\n    <script src=\"/js/i18n/datepicker.en.js\"></script>\n    <script src=\"/js/select2.min.js\"></script>\n    <script src=\"/js/moment-timezone.js\"></script>\n    <script src=\"/js/util.js\"></script>\n\n  </head>\n\n  <body>\n    <div class=\"container h-100\">\n      <div class=\"row\" id=\"container\">\n        <div class=\"col-lg-2 col-md-3 col-sm-4\" id=\"sidebar\">\n          <div id=\"fixedContainer\" class=\"pt-3\">\n              <h1><a href=\"/\">gath<span class=\"text-muted\">io</span></a></h1>\n          \n              <p class=\"lead text-center mb-4\">Nicer events</p>\n          \n              <a class=\"btn btn-success mb-2 btn-block\" href=\"/new/event/public\"><i class=\"far fa-calendar-plus\"></i> New event</a>\n          </div>        </div>\n        <div class=\"col-sm pt-3\" id=\"content\">\n          <div id=\"bodyContainer\">\n            <h1>404</h1>\n\n<p>Event not found!</p>\n\n<p>It may have never existed, or if it finished more than a week ago, it's been removed from the server. Don't despair - why not create a new one? I for one would love to come to your ocarina recital.</p>\n          </div>\n          <div id=\"footerContainer\">\n            <small class=\"text-muted\">\n              <a href=\"https://github.com/lowercasename/gathio\">GitHub</a> &middot; Made with <i class=\"far fa-heart\"></i> by <a href=\"https://raphaelkabo.com\">Raphael</a>.<br />\n            </small>\n          </div>\n        </div>\n      </div>\n    </div>\n  </body>\n</html>", headers: [{"content-length", "3592"}, {"content-type", "text/html; charset=utf-8"}, {"date", "Sun, 13 Nov 2022 23:17:53 GMT"}, {"etag", "W/\"e08-x7AJRScgGbgvCkPBa7bLzaoKV2A\""}, {"server", "Caddy"}, {"x-powered-by", "Express"}], method: :get, opts: [adapter: [follow_redirect: true, force_redirect: true, pool: :federation, connect_timeout: 5000, recv_timeout: 5000]], query: [], status: 404, url: "https://gath.io/.well-known/host-meta"}}
Nov 13 23:17:54 public.garden mix[24871]: 23:17:54.397 [debug] QUERY OK source="users" db=0.8ms idle=1475.8ms
Nov 13 23:17:54 public.garden mix[24871]: SELECT u0."id", u0."bio", u0."raw_bio", u0."email", u0."name", u0."nickname", u0."password_hash", u0."keys", u0."public_key", u0."ap_id", u0."avatar", u0."local", u0."follower_address", u0."following_address", u0."featured_address", u0."tags", u0."last_refreshed_at", u0."last_digest_emailed_at", u0."banner", u0."background", u0."note_count", u0."follower_count", u0."following_count", u0."is_locked", u0."is_confirmed", u0."password_reset_pending", u0."is_approved", u0."registration_reason", u0."confirmation_token", u0."default_scope", u0."domain_blocks", u0."is_active", u0."no_rich_text", u0."ap_enabled", u0."is_moderator", u0."is_admin", u0."show_role", u0."uri", u0."hide_followers_count", u0."hide_follows_count", u0."hide_followers", u0."hide_follows", u0."hide_favorites", u0."email_notifications", u0."mascot", u0."emoji", u0."pleroma_settings_store", u0."fields", u0."raw_fields", u0."is_discoverable", u0."invisible", u0."allow_following_move", u0."skip_thread_containment", u0."actor_type", u0."also_known_as", u0."inbox", u0."shared_inbox", u0."accepts_chat_messages", u0."last_active_at", u0."disclose_client", u0."accepts_email_list", u0."pinned_objects", u0."is_suggested", u0."last_status_at", u0."birthday", u0."show_birthday", u0."location", u0."language", u0."last_move_at", u0."notification_settings", u0."blocks", u0."mutes", u0."muted_reblogs", u0."muted_notifications", u0."subscribers", u0."multi_factor_authentication_settings", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."ap_id" = $1) ["https://gath.io/xJU4Zqvb33EKvATHChAbV"]
Nov 13 23:17:54 public.garden mix[24871]: 23:17:54.397 [debug] Fetching object https://gath.io/xJU4Zqvb33EKvATHChAbV via AP
Nov 13 23:17:54 public.garden mix[24871]: 23:17:54.478 [error] Could not decode user at fetch https://gath.io/xJU4Zqvb33EKvATHChAbV, {:content_type, "application/json; charset=utf-8"}
```

</details>

I also tried using the direct lookup URL, which is https://public.garden/@v84uEnHDkXI0DzxYBfyw4@gath.io; this also failed:

![Screenshot 2022-11-13 at 18-25-00 Public Garden](https://user-images.githubusercontent.com/987487/201550370-b29cb88a-8866-4a7b-9952-ef53c7211bf9.png)

The logs were a bit more verbose this time; I've (tried to) cut out all the non-related logs:

<details>

```
Nov 13 23:22:33 public.garden mix[24871]: 23:22:33.822 [info] GET /@v84uEnHDkXI0DzxYBfyw4@gath.io
Nov 13 23:22:33 public.garden mix[24871]: 23:22:33.824 request_id=FydIXZpvgVhPkg8AAjsh [debug] GET /@v84uEnHDkXI0DzxYBfyw4@gath.io
Nov 13 23:22:33 public.garden mix[24871]: 23:22:33.824 request_id=FydIXZpvgVhPkg8AAjsh [debug] Processing with Pleroma.Web.Feed.UserController.feed_redirect/2
Nov 13 23:22:33 public.garden mix[24871]:   Parameters: %{"nickname" => "v84uEnHDkXI0DzxYBfyw4@gath.io"}
Nov 13 23:22:33 public.garden mix[24871]:   Pipelines: [:accepts_html_xml_json, :http_signature, :static_fe]
Nov 13 23:22:33 public.garden mix[24871]: 23:22:33.825 request_id=FydIXZpvgVhPkg8AAjsh [debug] Sent 200 in 1ms
Nov 13 23:22:33 public.garden mix[24871]: 23:22:33.825 request_id=FydIXZpvgVhPkg8AAjsh [info] Sent 200 in 3ms
[...]
Nov 13 23:22:36 public.garden mix[24871]: 23:22:36.500 [info] GET /api/v1/accounts/lookup
Nov 13 23:22:36 public.garden mix[24871]: 23:22:36.501 request_id=FydIXjoI5M4_HrgAAjtR [debug] GET /api/v1/accounts/lookup
Nov 13 23:22:36 public.garden mix[24871]: 23:22:36.502 request_id=FydIXjoI5M4_HrgAAjtR [debug] Processing with Pleroma.Web.MastodonAPI.AccountController.lookup/2
Nov 13 23:22:36 public.garden mix[24871]:   Parameters: %{"acct" => "v84uEnHDkXI0DzxYBfyw4@gath.io"}
Nov 13 23:22:36 public.garden mix[24871]:   Pipelines: [:api]
Nov 13 23:22:36 public.garden mix[24871]: 23:22:36.503 request_id=FydIXjoI5M4_HrgAAjtR [debug] QUERY OK source="users" db=0.9ms queue=0.2ms idle=1582.0ms
Nov 13 23:22:36 public.garden mix[24871]: SELECT u0."id", u0."bio", u0."raw_bio", u0."email", u0."name", u0."nickname", u0."password_hash", u0."keys", u0."public_key", u0."ap_id", u0."avatar", u0."local", u0."follower_address", u0."following_address", u0."featured_address", u0."tags", u0."last_refreshed_at", u0."last_digest_emailed_at", u0."banner", u0."background", u0."note_count", u0."follower_count", u0."following_count", u0."is_locked", u0."is_confirmed", u0."password_reset_pending", u0."is_approved", u0."registration_reason", u0."confirmation_token", u0."default_scope", u0."domain_blocks", u0."is_active", u0."no_rich_text", u0."ap_enabled", u0."is_moderator", u0."is_admin", u0."show_role", u0."uri", u0."hide_followers_count", u0."hide_follows_count", u0."hide_followers", u0."hide_follows", u0."hide_favorites", u0."email_notifications", u0."mascot", u0."emoji", u0."pleroma_settings_store", u0."fields", u0."raw_fields", u0."is_discoverable", u0."invisible", u0."allow_following_move", u0."skip_thread_containment", u0."actor_type", u0."also_known_as", u0."inbox", u0."shared_inbox", u0."accepts_chat_messages", u0."last_active_at", u0."disclose_client", u0."accepts_email_list", u0."pinned_objects", u0."is_suggested", u0."last_status_at", u0."birthday", u0."show_birthday", u0."location", u0."language", u0."last_move_at", u0."notification_settings", u0."blocks", u0."mutes", u0."muted_reblogs", u0."muted_notifications", u0."subscribers", u0."multi_factor_authentication_settings", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."nickname" = $1) ["v84uEnHDkXI0DzxYBfyw4@gath.io"]
Nov 13 23:22:36 public.garden mix[24871]: 23:22:36.503 request_id=FydIXjoI5M4_HrgAAjtR [debug] Sent 404 in 2ms
Nov 13 23:22:36 public.garden mix[24871]: 23:22:36.503 request_id=FydIXjoI5M4_HrgAAjtR [info] Sent 404 in 3ms
[...]
Nov 13 23:26:57 public.garden mix[24871]: 23:26:57.260 [info] GET /api/v1/accounts/v84uEnHDkXI0DzxYBfyw4@gath.io
Nov 13 23:26:57 public.garden mix[24871]: 23:26:57.266 request_id=FydImvDOJir46yAAAj2h [debug] GET /api/v1/accounts/v84uEnHDkXI0DzxYBfyw4@gath.io
Nov 13 23:26:57 public.garden mix[24871]: 23:26:57.266 request_id=FydImvDOJir46yAAAj2h [debug] Processing with Pleroma.Web.MastodonAPI.AccountController.show/2
Nov 13 23:26:57 public.garden mix[24871]:   Parameters: %{"id" => "v84uEnHDkXI0DzxYBfyw4@gath.io"}
Nov 13 23:26:57 public.garden mix[24871]:   Pipelines: [:api]
Nov 13 23:26:57 public.garden mix[24871]: 23:26:57.267 request_id=FydImuSreoen-0EAAj1B [debug] QUERY OK source="instances" db=7.4ms idle=57.0ms
Nov 13 23:26:57 public.garden mix[24871]: SELECT i0."id", i0."host", i0."unreachable_since", i0."favicon", i0."favicon_updated_at", i0."inserted_at", i0."updated_at" FROM "instances" AS i0 WHERE (i0."host" = $1) ["mastodon.social"]
Nov 13 23:26:57 public.garden mix[24871]: 23:26:57.269 request_id=FydImuSreoen-0EAAj1B [debug] QUERY OK source="following_relationships" db=0.7ms queue=0.1ms idle=61.6ms
Nov 13 23:26:57 public.garden mix[24871]: SELECT u1."follower_address" FROM "following_relationships" AS f0 INNER JOIN "users" AS u1 ON f0."following_id" = u1."id" WHERE (f0."follower_id" = $1) AND (f0."state" = $2) ["APRwJAKymJaK19W22C", :follow_accept]
Nov 13 23:26:57 public.garden mix[24871]: 23:26:57.270 request_id=FydImuSreoen-0EAAj1B [debug] QUERY OK source="instances" db=0.3ms idle=62.5ms
Nov 13 23:26:57 public.garden mix[24871]: SELECT i0."id", i0."host", i0."unreachable_since", i0."favicon", i0."favicon_updated_at", i0."inserted_at", i0."updated_at" FROM "instances" AS i0 WHERE (i0."host" = $1) ["mastodon.social"]
Nov 13 23:26:57 public.garden mix[24871]: 23:26:57.271 request_id=FydImuSreoen-0EAAj1B [debug] QUERY OK source="following_relationships" db=0.5ms queue=0.1ms idle=59.9ms
Nov 13 23:26:57 public.garden mix[24871]: SELECT u1."follower_address" FROM "following_relationships" AS f0 INNER JOIN "users" AS u1 ON f0."following_id" = u1."id" WHERE (f0."follower_id" = $1) AND (f0."state" = $2) ["APRwJAKymJaK19W22C", :follow_accept]
Nov 13 23:26:57 public.garden mix[24871]: 23:26:57.272 request_id=FydImvDOJir46yAAAj2h [debug] QUERY OK source="oauth_tokens" db=4.8ms queue=0.2ms idle=62.7ms
Nov 13 23:26:57 public.garden mix[24871]: SELECT o0."id", o0."token", o0."refresh_token", o0."scopes", o0."valid_until", o0."user_id", o0."app_id", o0."inserted_at", o0."updated_at" FROM "oauth_tokens" AS o0 WHERE (o0."token" = $1) ["kNN909hiTUU5YjRP8kBWmHTrFYv0XNOGCYWlBPbUZ8M"]
Nov 13 23:26:57 public.garden mix[24871]: 23:26:57.274 [debug] QUERY OK source="users" db=1.2ms idle=58.0ms
Nov 13 23:26:57 public.garden mix[24871]: SELECT u0."id", u0."bio", u0."raw_bio", u0."email", u0."name", u0."nickname", u0."password_hash", u0."keys", u0."public_key", u0."ap_id", u0."avatar", u0."local", u0."follower_address", u0."following_address", u0."featured_address", u0."tags", u0."last_refreshed_at", u0."last_digest_emailed_at", u0."banner", u0."background", u0."note_count", u0."follower_count", u0."following_count", u0."is_locked", u0."is_confirmed", u0."password_reset_pending", u0."is_approved", u0."registration_reason", u0."confirmation_token", u0."default_scope", u0."domain_blocks", u0."is_active", u0."no_rich_text", u0."ap_enabled", u0."is_moderator", u0."is_admin", u0."show_role", u0."uri", u0."hide_followers_count", u0."hide_follows_count", u0."hide_followers", u0."hide_follows", u0."hide_favorites", u0."email_notifications", u0."mascot", u0."emoji", u0."pleroma_settings_store", u0."fields", u0."raw_fields", u0."is_discoverable", u0."invisible", u0."allow_following_move", u0."skip_thread_containment", u0."actor_type", u0."also_known_as", u0."inbox", u0."shared_inbox", u0."accepts_chat_messages", u0."last_active_at", u0."disclose_client", u0."accepts_email_list", u0."pinned_objects", u0."is_suggested", u0."last_status_at", u0."birthday", u0."show_birthday", u0."location", u0."language", u0."last_move_at", u0."notification_settings", u0."blocks", u0."mutes", u0."muted_reblogs", u0."muted_notifications", u0."subscribers", u0."multi_factor_authentication_settings", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."nickname" = $1) ["v84uEnHDkXI0DzxYBfyw4@gath.io"]
[...]
Nov 13 23:26:57 public.garden mix[24871]: 23:26:57.761 [warning] Can't find LRDD template in "https://gath.io/.well-known/host-meta": {:ok, %Tesla.Env{__client__: %Tesla.Client{adapter: {Tesla.Adapter.Hackney, :call, [[]]}, fun: nil, post: [], pre: []}, __module__: Tesla, body: "<!DOCTYPE html>\n<html>\n\n  <head>\n    <meta charset=\"utf-8\">\n    <meta name=\"viewport\" content=\"width=device-width, initial-scale=1, shrink-to-fit=no\">\n\n    <link rel=\"apple-touch-icon\" sizes=\"180x180\" href=\"/apple-touch-icon.png\">\n    <link rel=\"icon\" type=\"image/png\" sizes=\"32x32\" href=\"/favicon-32x32.png\">\n    <link rel=\"icon\" type=\"image/png\" sizes=\"16x16\" href=\"/favicon-16x16.png\">\n    <link rel=\"manifest\" href=\"/site.webmanifest\">\n    <link rel=\"mask-icon\" href=\"/safari-pinned-tab.svg\" color=\"#5bbad5\">\n    <meta name=\"msapplication-TileColor\" content=\"#da532c\">\n    <meta name=\"theme-color\" content=\"#ffffff\">\n\n    <meta property=\"og:type\" content=\"website\">\n    <meta property=\"og:image:width\" content=\"260\">\n    <meta property=\"og:image:height\" content=\"260\">\n    <meta property=\"og:description\" content=\"An easier, quicker, and much less privacy-invading way to make and share events\">\n    <meta property=\"og:title\" content=\"\">\n    <meta property=\"og:url\" content=\"https:///\">\n    <meta property=\"og:image\" content=\"https:///og-image.jpg\">\n\n    <meta name=\"twitter:card\" content=\"summary\">\n    <meta name=\"twitter:title\" content=\"\">\n    <meta name=\"twitter:description\" content=\"An easier, quicker, and much less privacy-invading way to make and share events\">\n    <meta name=\"twitter:image\" content=\"https:///og-image.jpg\">\n\n\n    <title></title>\n\n    <!-- FONTS -->\n    <link rel=\"stylesheet\" href=\"/css/fontawesome.css\">\n\n    <!-- CSS -->\n    <link rel=\"stylesheet\" href=\"/css/bootstrap.min.css\">\n    <link href=\"/css/datepicker.min.css\" rel=\"stylesheet\" type=\"text/css\">\n    <link rel=\"stylesheet\" href=\"/css/select2.min.css\">\n    <link rel=\"stylesheet\" href=\"/css/style.css\">\n\n    <!-- JS -->\n    <script src=\"/js/jquery-3.4.1.min.js\"></script>\n    <script src=\"/js/popper.min.js\"></script>\n    <script src=\"/js/jquery.form-validator.js\"></script>\n    <script src=\"/js/bootstrap.min.js\"></script>\n    <script src=\"/js/datepicker.min.js\"></script>\n    <script src=\"/js/moment.js\"></script>\n    <script src=\"/js/jquery.uploadPreview.min.js\"></script>\n    <script src=\"/js/clipboard.min.js\"></script>\n    <script src=\"/js/autosize.min.js\"></script>\n    <script src=\"/js/i18n/datepicker.en.js\"></script>\n    <script src=\"/js/select2.min.js\"></script>\n    <script src=\"/js/moment-timezone.js\"></script>\n    <script src=\"/js/util.js\"></script>\n\n  </head>\n\n  <body>\n    <div class=\"container h-100\">\n      <div class=\"row\" id=\"container\">\n        <div class=\"col-lg-2 col-md-3 col-sm-4\" id=\"sidebar\">\n          <div id=\"fixedContainer\" class=\"pt-3\">\n              <h1><a href=\"/\">gath<span class=\"text-muted\">io</span></a></h1>\n          \n              <p class=\"lead text-center mb-4\">Nicer events</p>\n          \n              <a class=\"btn btn-success mb-2 btn-block\" href=\"/new/event/public\"><i class=\"far fa-calendar-plus\"></i> New event</a>\n          </div>        </div>\n        <div class=\"col-sm pt-3\" id=\"content\">\n          <div id=\"bodyContainer\">\n            <h1>404</h1>\n\n<p>Event not found!</p>\n\n<p>It may have never existed, or if it finished more than a week ago, it's been removed from the server. Don't despair - why not create a new one? I for one would love to come to your ocarina recital.</p>\n          </div>\n          <div id=\"footerContainer\">\n            <small class=\"text-muted\">\n              <a href=\"https://github.com/lowercasename/gathio\">GitHub</a> &middot; Made with <i class=\"far fa-heart\"></i> by <a href=\"https://raphaelkabo.com\">Raphael</a>.<br />\n            </small>\n          </div>\n        </div>\n      </div>\n    </div>\n  </body>\n</html>", headers: [{"content-length", "3592"}, {"content-type", "text/html; charset=utf-8"}, {"date", "Sun, 13 Nov 2022 23:26:57 GMT"}, {"etag", "W/\"e08-x7AJRScgGbgvCkPBa7bLzaoKV2A\""}, {"server", "Caddy"}, {"x-powered-by", "Express"}], method: :get, opts: [adapter: [follow_redirect: true, force_redirect: true, pool: :federation, connect_timeout: 5000, recv_timeout: 5000]], query: [], status: 404, url: "https://gath.io/.well-known/host-meta"}}
Nov 13 23:26:57 public.garden mix[24871]: 23:26:57.846 [debug] QUERY OK source="users" db=0.9ms idle=599.0ms
Nov 13 23:26:57 public.garden mix[24871]: SELECT u0."id", u0."bio", u0."raw_bio", u0."email", u0."name", u0."nickname", u0."password_hash", u0."keys", u0."public_key", u0."ap_id", u0."avatar", u0."local", u0."follower_address", u0."following_address", u0."featured_address", u0."tags", u0."last_refreshed_at", u0."last_digest_emailed_at", u0."banner", u0."background", u0."note_count", u0."follower_count", u0."following_count", u0."is_locked", u0."is_confirmed", u0."password_reset_pending", u0."is_approved", u0."registration_reason", u0."confirmation_token", u0."default_scope", u0."domain_blocks", u0."is_active", u0."no_rich_text", u0."ap_enabled", u0."is_moderator", u0."is_admin", u0."show_role", u0."uri", u0."hide_followers_count", u0."hide_follows_count", u0."hide_followers", u0."hide_follows", u0."hide_favorites", u0."email_notifications", u0."mascot", u0."emoji", u0."pleroma_settings_store", u0."fields", u0."raw_fields", u0."is_discoverable", u0."invisible", u0."allow_following_move", u0."skip_thread_containment", u0."actor_type", u0."also_known_as", u0."inbox", u0."shared_inbox", u0."accepts_chat_messages", u0."last_active_at", u0."disclose_client", u0."accepts_email_list", u0."pinned_objects", u0."is_suggested", u0."last_status_at", u0."birthday", u0."show_birthday", u0."location", u0."language", u0."last_move_at", u0."notification_settings", u0."blocks", u0."mutes", u0."muted_reblogs", u0."muted_notifications", u0."subscribers", u0."multi_factor_authentication_settings", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."ap_id" = $1) ["https://gath.io/v84uEnHDkXI0DzxYBfyw4"]
Nov 13 23:26:57 public.garden mix[24871]: 23:26:57.846 [debug] Fetching object https://gath.io/v84uEnHDkXI0DzxYBfyw4 via AP
Nov 13 23:26:57 public.garden mix[24871]: 23:26:57.935 [error] Could not decode user at fetch https://gath.io/v84uEnHDkXI0DzxYBfyw4, {:content_type, "application/json; charset=utf-8"}
Nov 13 23:26:57 public.garden mix[24871]: 23:26:57.935 request_id=FydImvDOJir46yAAAj2h [debug] Sent 404 in 669ms
Nov 13 23:26:57 public.garden mix[24871]: 23:26:57.935 request_id=FydImvDOJir46yAAAj2h [info] Sent 404 in 675ms
```

</details>


## Analysis

Two obvious errors stick out:

> Can't find LRDD template in "https://gath.io/.well-known/host-meta"

followed by a copy of the index page as HTML; presumably Pleroma is expecting JSON on that URL

And:

> Fetching object https://gath.io/v84uEnHDkXI0DzxYBfyw4 via AP
> Could not decode user at fetch https://gath.io/v84uEnHDkXI0DzxYBfyw4, {:content_type, "application/json; charset=utf-8"}

Maybe this LRDD error is a red-herring. The "could not decode" definitely looks like the crux of the problem.

## Future Work

My next plan will be to start turning up the logging on and maybe even editing my own Gathio server to try to see the actual ActivityPub responses its sending. Hopefully whatever malformation will stick out. Or maybe Gath is doing the right thing and its Pleroma that is being unnecessarily picky, in which case I'll try to get it patched over there.

**Comment by lowercasename** [comment]
I was wondering if it could be related to this Pleroma issue: https://github.com/wmurphyrd/guppe/issues/32

But the route handler for an event responds with an AP JSON object if the `Accept` header is either `application/activity+json` _or_ `application/json` - I'm not sure whether (or even if) Pleroma is prioritising the `application/json` response in some way and failing to parse it at that point.

Here's the JSON response for that event page, for reference:

```
{
    "@context": [
        "https://www.w3.org/ns/activitystreams",
        "https://w3id.org/security/v1"
    ],
    "id": "https://gath.io/v84uEnHDkXI0DzxYBfyw4",
    "type": "Person",
    "preferredUsername": "v84uEnHDkXI0DzxYBfyw4",
    "inbox": "https://gath.io/activitypub/inbox",
    "outbox": "https://gath.io/v84uEnHDkXI0DzxYBfyw4/outbox",
    "followers": "https://gath.io/v84uEnHDkXI0DzxYBfyw4/followers",
    "summary": "<p><p>sdjkfhsduifhsduifhuisdhfuisdhfuisdh</p>\n</p><p>Location: sdiufhsduifhsduifhsduifhsduihfuish.</p><p>Starting 25 November 2022 6:21 pm America/Toronto.</p>",
    "name": "sdouifhsduiofhsduihfuish",
    "featured": "https://gath.io/v84uEnHDkXI0DzxYBfyw4/featured",
    "publicKey": {
        "id": "https://gath.io/v84uEnHDkXI0DzxYBfyw4#main-key",
        "owner": "https://gath.io/v84uEnHDkXI0DzxYBfyw4",
        "publicKeyPem": "-----BEGIN PUBLIC KEY-----\nMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAo8ACm0YIUwBMkz4B/i86\nRnVuiHTKytXeZGZ+gIJhgh5ZtSJmWNB7eDKFz4N2JHa/ez6H57Z9WDMCnrZNyJM0\nDg63rKC6gJLbVmkGiPAzDwDVy0ZIVYWb+xzQ3QK1NQ1ZcHKSbnRZu8IaanGcJG9N\ndrtSdPg3/rImxxkTX++U3LynuHNMp3UEM/l2O2m49tgbZlK2BUA/0p1RppHMeeRl\n2AtUPVnrkZGtQD5KjQ5UbSe+VQeJfugTxrDnjuajZy+ojrymBimUJqg961Wjr7WQ\nri77EK+QexsKzPNbI+wbdY36JP7IlONVC4H5DO6gFk6acrzim+AQZkhkpO3chWPp\n/wIDAQAB\n-----END PUBLIC KEY-----\n"
    }
}
```

**Comment by Michcioperz** [comment]
Regarding the host-meta endpoint, Pleroma/Akkoma expects an XML or JSON there describing the location of the webfinger endpoint, but it seems to fallback to the default location. https://akkoma.dev/AkkomaGang/akkoma/src/commit/9deae8c533f1579f3491df41ab08697e9c6589b6/lib/pleroma/web/web_finger.ex

As for the second noticed issue, it seems to trip up at the unfamiliar mimetype? https://akkoma.dev/AkkomaGang/akkoma/src/commit/9deae8c533f1579f3491df41ab08697e9c6589b6/lib/pleroma/object/fetcher.ex#L270

**Comment by raphaelkabo** [comment]
Oh, I see what the issue might be - I'm sending `application/json` rather than `application/activity+json` in the response. I'll make a new branch to see what happens if I update that!

**Comment by lowercasename** [comment]
Oh, I see what the issue might be - I'm sending application/json rather than application/activity+json in the response. I'll make a new branch to see what happens if I update that!

**Comment by lowercasename** [comment]
@kousu, I've set up an instance with the new branch on https://dev.gath.io - could you see if you have any more success with Pleroma there? You'll need to create a new event as the old event will still be linked to the gath.io domain.

**Comment by dariusk** [comment]
Thanks for this great bug report. I'll take some time in the next week to try and figure this out. I'll reinstall my pleroma test server and see what I can see.

**Comment by lowercasename** [comment]
I've been looking into this today and yesterday. I've got a testing instance where I've fixed the communication between Pleroma and Gathio when it comes to polls (thanks for the bang-on pointers, @kousu), so now my Pleroma user can successfully reply to the poll coming from the Gathio event. However, I still can't get the Pleroma user to appear in the event's attendees, mysteriously. That'll need some more debugging. But good progress so far on this very long-standing issue!
